diff a/source/host/release/release/common_insert.sh b/source/host/release/common_insert.sh
--- a/source/host/release/common_insert.sh	
+++ b/source/host/release/common_insert.sh	
@@ -25,7 +25,7 @@
 #							13   WLAN STATION + BT CLASSIC MODE + BT LE MODE
 #							14   WLAN ACCESS POINT + BT CLASSIC MODE+ BT LE MODE
 
-COEX_MODE=1
+COEX_MODE=13
 
 #To enable TA-level SDIO aggregation set 1 else set 0 to disable it.
 TA_AGGR=8
